daemon/logger/loggerutils/rotatefilewriter.go

```diff
@@ -42,6 +44,9 @@ func NewRotateFileWriter(logPath string, capacity int64, maxFiles int) (*RotateF
 //WriteLog write log message to File
 func (w *RotateFileWriter) Write(message []byte) (int, error) {
 	w.mu.Lock()
+	if w.closed {
+		return -1, errors.New("cannot write because the output file was closed")
```

Comment:
This returns without unlocking.

I suggest using `defer` in this function as well.